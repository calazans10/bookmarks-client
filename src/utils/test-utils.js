import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import rootReducer from '../reducers';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
