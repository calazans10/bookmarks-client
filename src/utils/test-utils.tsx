import React, { ReactNode } from 'react';
import { AnyAction, Action, createStore, Store } from 'redux';
import { Provider } from 'react-redux';
import { render, RenderResult } from '@testing-library/react';
import rootReducer from '../reducers';

interface RenderWithRedux<
  S = any,
  A extends Action = AnyAction,
  I extends S = any
  > {
  (
    ui: ReactNode,
    reduxOptions?: {
      store?: Store<S, A>
      initialState?: I
    }
  ): RenderResult & {
    store: Store<S, A>
  }
}

export const renderWithRedux: RenderWithRedux = (
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
