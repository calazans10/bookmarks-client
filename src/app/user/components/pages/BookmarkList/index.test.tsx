import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithRedux } from 'test-utils';
import { BookmarkList } from './index';

describe('BookmarkList', () => {
  it('renders without crashing', () => {
    const { container } = renderWithRedux(
      <MemoryRouter>
        <BookmarkList
          count={1}
          offset={1}
          limit={10}
          total={10}
          onChangeMyBookmarksMeta={jest.fn()}
          onRequestGetMyBookmarks={jest.fn()}
        />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
