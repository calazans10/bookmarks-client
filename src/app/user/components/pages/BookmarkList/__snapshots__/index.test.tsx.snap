// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkList renders without crashing 1`] = `
<div
  class="sc-AxheI iAQAMi"
>
  <div
    class="sc-AxiKw bJCLjj"
  >
    <p
      class="sc-AxhCb dvaKgx"
    >
      <span
        class="sc-AxhUy funJuC"
      />
       
      
    </p>
    <button
      aria-label="Close alert"
      class="sc-AxgMl bMsCBu"
      type="button"
    >
      <span>
        Close
      </span>
    </button>
  </div>
  <header
    class="sc-Axmtr eCCXJi"
  >
    <div
      class="sc-AxmLO hiBYtb"
    >
      <div
        class="sc-AxjAm daLWxe"
      >
        <a
          class="sc-AxirZ gHWAZY"
          href="/bookmarks"
        >
          Bookmarks
        </a>
      </div>
      <nav
        class="sc-fzozJi jmrGBi"
      >
        <ul>
          <li
            class="sc-fzoLsD jBEMxY"
          >
            <button
              class="sc-fzpans joGuf"
              type="button"
            >
              Sign Out
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <main
    class="sc-fzplWN fiNEmJ"
  >
    <div
      class="sc-fznyAO kxqtUh"
    >
      <ul>
        <li>
          <a
            href="/bookmarks/new"
          >
            Add bookmark
          </a>
        </li>
      </ul>
    </div>
    <table
      class="sc-fznZeY MAyRa"
    >
      <tbody />
    </table>
  </main>
</div>
`;
